src/python/pants/backend/python/rules/pex_from_target_closure.py

```diff
@@ -36,7 +31,7 @@ class CreatePexFromTargetClosure:
 @rule(name="Create PEX from targets")
 async def create_pex_from_target_closure(
     request: CreatePexFromTargetClosure, python_setup: PythonSetup
-) -> Pex:
+) -> CreatePex:
```

Comment:
Why make this change? It's simpler for a call site to get back a `Pex` than `CreatePex`, which is a request type.

Reply:
A call site does absolutely nothing different here. The difference this makes is that it enables getting a `CreatePex` from a target closure, then modifying it, which is what #8940 does once I push the most recent changes. Since this doesn't change anything at all, but makes the rule graph a little more open (in the same way as the hermetic pex rules now produce an `ExecuteProcessRequest` directly), I would prefer to keep it.

Reply:
Ah, so to confirm, the user can still say `await Get[Pex](CreatePexFromTargetClosure) and then the engine will automatically convert this resulting `CreatePex -> Pex`? Cool! This sounds great then!